src/ivc/cyclefold/sfc/input/assigned.rs

```diff
@@ -660,69 +884,58 @@ impl<const A: usize, F: PrimeField> Input<A, F> {
         );
 
         let mg = MainGate::new(main_gate_config.clone());
-        let is_zero_term = mg.is_zero_term(region, self.step.clone())?;
+        let is_zero_step = mg.is_zero_term(region, self.step.clone())?;
 
         let zero = region.assign_advice(|| "", main_gate_config.state[0], Value::known(F::ZERO))?;
         region.next();
 
-        let expected_l0 = mg.conditional_select(region, &zero, &poly_L_values[0], &is_zero_term)?;
-        let expected_l0 = bn_chip
+        let expected_l0 = mg.conditional_select(region, &zero, &poly_L_values[0], &is_zero_step)?;
+        let expected_l0_limbs = bn_chip
             .from_assigned_value_to_limbs(region, &expected_l0)
             .map_err(|err| {
                 error!("while make from L0 biguint form: {err:?}");
                 Halo2PlonkError::Synthesis
             })?;
 
-        let expected_l1 = mg.conditional_select(region, &zero, &poly_L_values[1], &is_zero_term)?;
-        let expected_l1 = bn_chip
+        let expected_l1 = mg.conditional_select(region, &zero, &poly_L_values[1], &is_zero_step)?;
+        let expected_l1_limbs = bn_chip
             .from_assigned_value_to_limbs(region, &expected_l1)
             .map_err(|err| {
                 error!("while make from L1 biguint form: {err:?}");
                 Halo2PlonkError::Synthesis
             })?;
 
-        for (acc_W, incoming_W, trace, new_acc_W) in itertools::multizip((
+        for (acc_W, incoming_W, trace, new_acc_W, index) in itertools::multizip((
```

Comment:
It might be a good idea to add a comment of how many delegation of ecc multiplication have been used somewhere. maybe here or in function comment

Reply:
Ok, will add a comment in future PR